src/m365/commands/setup.ts

```diff
@@ -106,59 +205,218 @@ class SetupCommand extends AnonymousCommand {
-        settings = this.getSettings(preferences);
-      }
+    if (!preferences.summary) {
+      return;
+    }
 
+    // used only for testing. Normally, we'd get the settings from the answers
+    /* c8 ignore next 3 */
+    if (!settings) {
+      settings = this.getSettings(preferences);
+    }
+
+    await logger.logToStderr('');
+    await logger.logToStderr('Configuring settings...');
+    await logger.logToStderr('');
+
+    await this.configureSettings({ preferences, settings, silent: false, logger });
+
+    if (!this.verbose) {
       await logger.logToStderr('');
-      await logger.logToStderr('Configuring settings...');
-      await logger.logToStderr('');
+      await logger.logToStderr(chalk.green('DONE'));
+    }
+  }
 
-      await this.configureSettings(settings, false, logger);
+  private async configureExistingEntraApp(logger: Logger): Promise<Preferences> {
+    await logger.logToStderr('Please provide the details of the existing app registration.');
+    let clientCertificateFile: string | undefined;
+    let clientCertificateBase64Encoded: string | undefined;
+    let clientCertificatePassword: string | undefined;
 
-      if (!this.verbose) {
-        await logger.logToStderr('');
-        await logger.logToStderr(chalk.green('DONE'));
+    const clientId = await cli.promptForInput({
+      message: 'Client ID:',
+      /* c8 ignore next */
+      validate: value => value.length > 0
+    });
+    const tenantId = await cli.promptForInput({
+      message: 'Tenant ID (leave common if the app is multitenant):',
+      default: 'common'
+    });
+    const clientSecret = await cli.promptForInput({
+      message: 'Client secret (leave empty if you use a certificate or a public client):'
+    });
+    if (!clientSecret) {
+      clientCertificateFile = await cli.promptForInput({
+        message: `Path to the client certificate file (leave empty if you want to specify a base64-encoded certificate string):`
+      });
+      if (!clientCertificateFile) {
+        clientCertificateBase64Encoded = await cli.promptForInput({
+          message: 'Base64-encoded certificate string:'
+        });
       }
+      if (clientCertificateFile || clientCertificateBase64Encoded) {
+        clientCertificatePassword = await cli.promptForInput({
+          message: 'Password for the client certificate (leave empty if the certificate is not password-protected):'
+        });
+      }
+    }
+
+    return {
+      clientId,
+      tenantId,
+      clientSecret,
+      clientCertificateFile,
+      clientCertificateBase64Encoded,
+      clientCertificatePassword
+    };
+  }
+
+  private async createNewEntraApp(preferences: Preferences, logger: Logger): Promise<AppInfo> {
+    if (!await cli.promptForConfirmation({
+      message: 'CLI for Microsoft 365 will now sign in to your Microsoft 365 tenant as Microsoft Azure CLI to create a new app registration. Continue?',
+      default: false
+    })) {
+      throw new Error();
     }
```

Comment:
Yes, I agree with this @waldekmastykarz. It seems more logical if the default is `true`

Reply:
I made it false on purpose, so that you don't automatically get an Entra app in an incorrect tenant because you pressed enter accidentally. I suggest we keep false and make it very deliberate for folks to connect.

Reply:
Isn't it kinda deliberately already? I mean we kinda ask twice already about that.
Also still if we select No it should not present an error rather it should just end

Reply:
To me, creating an Entra app reg in a tenant is a serious thing, hence I think asking twice is good, just to be sure and to properly manage expectations. Agreed about closing but not sure if that's not a bigger issue (ie. how we implement prompts) rather than something specific to this feature.

Reply:
How about this:

<img width="1032" alt="image" src="https://github.com/user-attachments/assets/350d7bba-9acf-4dee-b76a-a8b3ee826591">

It shows an error because of how we handle exceptions and promises. The only alternative is to immediately close the process but not sure if that's desired.

Reply:
> How about this:
> 
> <img alt="image" width="1032" src="https://private-user-images.githubusercontent.com/11164679/362675948-350d7bba-9acf-4dee-b76a-a8b3ee826591.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5MzM5NzcsIm5iZiI6MTcyNDkzMzY3NywicGF0aCI6Ii8xMTE2NDY3OS8zNjI2NzU5NDgtMzUwZDdiYmEtOWFjZi00ZGVlLWI3NmEtYThiM2VlODI2NTkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI5VDEyMTQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiOGYxZGQ1MTE5MDY4NzVjMmJkNmJkNjAwMzYyMTM4YTJiM2FhZTYyNjk4YTZiYjU0MTZmZjhjNjBlYjJkMjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Jskc-cf36rdx5UaQXcSyeFce1jXaU_1U0Qf0KZfhA3c">
> It shows an error because of how we handle exceptions and promises. The only alternative is to immediately close the process but not sure if that's desired.

ok clear. Yes that works for me 👍

Reply:
> To me, creating an Entra app reg in a tenant is a serious thing, hence I think asking twice is good, just to be sure and to properly manage expectations. Agreed about closing but not sure if that's not a bigger issue (ie. how we implement prompts) rather than something specific to this feature.

yes I agree that asking twice is a good thing. I just don't quite feel that the second question by default is No not Yes 🤔.
But if you are serious about then it may stay. I don't have any strong opinion on this subject this is just what I feel and the way I would code it (yes as default)

Reply:
I'm cautious about us creating something in the wrong tenant that might lead to folks having to clean it up. Measure twice, cut once 😅

Reply:
Let's keep in mind that the decision we take today is not set in stone. If we ship this as-is, and in a month or two learn that folks prefer Y to be default, we can adjust it without any bigger consequences.